.github/workflows/linux.yml

```diff
@@ -0,0 +1,176 @@
+      - master
+      - 'releases/**'
+    paths-ignore:
+      - 'modules/nvidia_plugin'
+      - 'modules/openvino_code'
+
+concurrency:
+  # github.ref is not unique in post-commit
+  group: ${{ github.event_name == 'push' && github.run_id || github.ref }}-linux
+  cancel-in-progress: true
+
+env:
+  PYTHON_VERSION: '3.11'
+
+jobs:
+  Build_and_test:
+    name: Build and Test
+    timeout-minutes: 150
+    defaults:
+      run:
+        shell: bash
+    runs-on: ubuntu-20.04-16-cores
+    container:
+      image: ubuntu:20.04
+    env:
+      DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
+      CMAKE_BUILD_TYPE: 'Release'
+      CMAKE_CXX_COMPILER_LAUNCHER: ccache
+      CMAKE_C_COMPILER_LAUNCHER: ccache
+      GITHUB_WORKSPACE: /__w/openvino_contrib/openvino_contrib
+      OPENVINO_REPO: /__w/openvino_contrib/openvino_contrib/openvino
+      OPENVINO_CONTRIB_REPO: /__w/openvino_contrib/openvino_contrib/openvino_contrib
+      TEST_DATA: /__w/openvino_contrib/openvino_contrib/testdata
+      INSTALL_DIR: /__w/openvino_contrib/openvino_contrib/openvino_install
+      BUILD_DIR: /__w/openvino_contrib/openvino_contrib/openvino_build
+      GRADLE_VER: '7.1.1'
+
+    steps:
+      - name: Set apt retries
+        run: echo 'Acquire::Retries "10";' > /etc/apt/apt.conf.d/80-retries
+
+      - name: Install git
+        run: |
+          apt-get update
+          apt-get install --assume-yes --no-install-recommends git git-lfs ca-certificates
+
+      - name: Clone OpenVINO
+        uses: actions/checkout@v4
+        with:
+          repository: 'openvinotoolkit/openvino'
+          path: ${{ env.OPENVINO_REPO }}
+          submodules: 'true'
+          ref: 'master'
+
+      - name: Clone OpenVINO Contrib
+        uses: actions/checkout@v4
+        with:
+          path: ${{ env.OPENVINO_CONTRIB_REPO }}
+          submodules: 'true'
+      
+      - name: Clone Testdata
+        uses: actions/checkout@v4
+        with:
+          repository: 'openvinotoolkit/testdata'
+          path: ${{ env.TEST_DATA }}
+          lfs: 'true'
+          submodules: 'true'
+
+      #
+      # Dependencies
+      #
+
+      - name: Install build dependencies
+        run: |
+          bash ${OPENVINO_REPO}/install_build_dependencies.sh
+          # default-jdk - Java API; unzip for gradle installation
+          apt install --assume-yes --no-install-recommends default-jdk libopencv-dev unzip
+      
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          gradle-version: ${{ env.GRADLE_VER }}
+    
+      - name: Setup Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install python dependencies
+        run: python3 -m pip install -r ${OPENVINO_REPO}/src/bindings/python/wheel/requirements-dev.txt
+          
+      - name: Setup ccache
+        uses: hendrikmuhs/ccache-action@v1.2
```

Comment:
lets use gha cache action instead

Reply:
You would need to set environment variable CCACHE_MAXSIZE for that only

Reply:
Why use the `cache` actions directly? Using `hendrikmuhs/ccache-action` removes the need to install and configure `ccache` and it handles the packing, directories and uploading/downloading processes.

Reply:
to reduce dependencies on 3rd party component, it may help only on windows, cmake on linux/mac should be installed by install dependencies script. But imho we can install it on windows too

Reply:
Which other steps, except installing ccache, we need to implement on our side if we use gha action instead of this one?

By the way, I guess, if the only concern is a possibility of sudden breaking changes made to a custom action, we can restrict its version to a specific commit

Reply:
As discussed with @dorloff and @mryzhov, the GH-provided action will be used -> no dependency on the unknown third-party, `actions/cache` is supposed to be more reliable.